lib/internal/streams/operators.js

```diff
@@ -92,9 +93,35 @@ function map(fn, options) {
     let next;
     let resume;
     let done = false;
+    let cnt = 0;
 
-    function onDone() {
+    const hwm = concurrency * 2 - 1;
+
+    function onThen() {
+      maybeResume();
```

Comment:
```suggestion
```

Reply:
I instead renamed the `onFinally` and remove the `onThen`